js/ui/codeView.js

```diff
@@ -96,8 +98,16 @@ const _ensureHackDataFile = (function () {
             return keyfile;
 
         try {
-            keyfile.load_from_dirs('hack-data.ini', searchPaths,
+            const [, path] = keyfile.load_from_dirs('hack-data.ini', searchPaths,
                 GLib.KeyFileFlags.NONE);
+            monitorFile = Gio.file_new_for_path(path);
+            monitor = monitorFile.monitor_file(Gio.FileMonitorFlags.NONE, null);
+            monitor.connect('changed', (_monitor, _file, _otherfile, ev, _data) => {
```

Comment:
Shouldn’t this signal handler be disconnected at some point?

Reply:
This will exists until the end of the gnome shell, has the same scope as the keyfile

Reply:
We discussed this a bit on the desktop daily. I agree it is good practice to disconnect signal handlers, even if they will last for the lifetime of an application. However, in this case we'll probably be okay: the callback just assigns to some variables in its closure, and we assume that gjs will keep that closure alive until process termination. I guess we'll find out the hard way if we're wrong.